### PR TITLE
Separate service in registry for NSMgr 

### PR DESCRIPTION
**Pull request by lobkovilya**
Signed-off-by: Ilya Lobkov <lobkovilya@yandex.ru>

## Description
According to https://github.com/networkservicemesh/networkservicemesh/issues/799 method UpdateNSM has been moved to separate NsmRegistryService and became more general method called RegisterNSM. Also add method for fetching endpoints by NSMgr name - GetEndpoints.

## How Has This Been Tested?

- [X] Covered by existing integration testing
- [X] Added integration testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/pkg/tests/nsmd_test_utils.go [hunk 1]
```diff
@@ -31,69 +31,94 @@ import (
...
+}
+
 type nsmdTestServiceDiscovery struct {
 	apiRegistry *testApiRegistry
-
-	services   map[string]*registry.NetworkService
-	managers   map[string]*registry.NetworkServiceManager
-	endpoints  map[string]*registry.NetworkServiceEndpoint
-	nsmCounter int
+	storage     *sharedStorage
```
Comment (nickolaev) [inline controlplane/pkg/tests/nsmd_test_utils.go]: Nit: is this really a storage?

File: controlplane/pkg/tests/nsmd_test_utils.go [hunk 2]
```diff
@@ -31,69 +31,94 @@ import (
...
-func newNSMDTestServiceDiscovery(testApi *testApiRegistry) *nsmdTestServiceDiscovery {
+func newNSMDTestServiceDiscovery(testApi *testApiRegistry, nsmName string, storage *sharedStorage) *nsmdTestServiceDiscovery {
 	return &nsmdTestServiceDiscovery{
-		services:    make(map[string]*registry.NetworkService),
-		endpoints:   make(map[string]*registry.NetworkServiceEndpoint),
-		managers:    make(map[string]*registry.NetworkServiceManager),
+		storage: storage,
+		//services:    make(map[string]*registry.NetworkService),
+		//endpoints:   make(map[string]*registry.NetworkServiceEndpoint),
+		//managers:    make(map[string]*registry.NetworkServiceManager),
```
Comment (nickolaev) [inline controlplane/pkg/tests/nsmd_test_utils.go]: Please delete these.

File: controlplane/pkg/tests/nsmd_test_utils.go [hunk 3]
```diff
@@ -234,14 +259,19 @@ func (impl *nsmdTestServiceRegistry) GetPublicAPI() string {
...
 }
 
-func (impl *nsmdTestServiceRegistry) RegistryClient() (registry.NetworkServiceRegistryClient, error) {
+func (impl *nsmdTestServiceRegistry) NseRegistryClient() (registry.NetworkServiceRegistryClient, error) {
 	return impl.nseRegistry, nil
 }
 
+func (impl *nsmdTestServiceRegistry) NsmRegistryClient() (registry.NsmRegistryClient, error) {
+	return impl.nseRegistry, nil
+
```
Comment (nickolaev) [inline controlplane/pkg/tests/nsmd_test_utils.go]: Nit: extra empty line.

File: controlplane/pkg/tests/nsmd_test_utils.go [hunk 4]
```diff
@@ -323,18 +353,14 @@ func (impl *nsmdFullServerImpl) addFakeDataplane(dp_name string, dp_addr string)
 func (srv *nsmdFullServerImpl) registerFakeEndpoint(networkServiceName string, payload string, nse_address string) *registry.NSERegistration {
 	return srv.registerFakeEndpointWithName(networkServiceName, payload, nse_address, networkServiceName+"provider")
 }
-func (srv *nsmdFullServerImpl) registerFakeEndpointWithName(networkServiceName string, payload string, nse_address string, endpointname string) *registry.NSERegistration {
+func (srv *nsmdFullServerImpl) registerFakeEndpointWithName(networkServiceName string, payload string, nsmName string, endpointname string) *registry.NSERegistration {
```
Comment (nickolaev) [inline controlplane/pkg/tests/nsmd_test_utils.go]: Nit: could we use `nsmgrName`, or maybe `mgrName` something that distinguish us from the more generic `nsm`.

File: k8s/pkg/registryserver/helpers.go [hunk 5]
```diff
@@ -36,3 +36,14 @@ func mapNsmFromCustomResource(cdr *v1.NetworkServiceManager) *registry.NetworkSe
 		LastSeen: lastSeen,
 	}
 }
+
+func mapNseFromCustomResource(cdr *v1.NetworkServiceEndpoint, payload string) *registry.NetworkServiceEndpoint {
```
Comment (nickolaev) [inline k8s/pkg/registryserver/helpers.go]: Question: what `cdr` stands for?

Comment (lobkovilya) [reply k8s/pkg/registryserver/helpers.go]: I meant Custom Defined Resource, but I think just 'cr' is more preferable. What do you think?

Comment (nickolaev) [reply k8s/pkg/registryserver/helpers.go]: Both crd and cr work for me.

File: controlplane/pkg/tests/nsmd_test_utils.go (hunk 1 above)
Comment (lobkovilya) [reply controlplane/pkg/tests/nsmd_test_utils.go]: This is class that stores all custom resources for all service registries, so I think storage is appropriate name

**Pull request merged**